libs/sdk-core/src/error.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Error;
+use std::fmt;
+
+pub type SdkResult<T, E = NewSdkError> = Result<T, E>;
+
+/// Type of error returned by the SDK
+#[derive(Debug)]
+pub enum NewSdkError {
+    SdkFailedToRequestBackup,
```

Comment:
A few comments regarding all error variants:
1. No need to prefix with sdk.
2. better to use the verb at the end, e.g: BackupRequestFailed
3. Worth adding a comment on each error that describes it.
4. Missing error message, worth considering using thisError as uniffi suggests: https://mozilla.github.io/uniffi-rs/udl/errors.html

Reply:
Done